### PR TITLE
Support nested sidebar items in the docs

### PR DESCRIPTION
**Pull request by dtuite**
Preview: https://deploy-preview-1011--roadie.netlify.app/docs/

Demo loom:

https://www.loom.com/share/08cda352e2244ecf834742c17eca634a

### CODE REVIEW

**Comment by dtuite** [comment]
https://app.shortcut.com/larder/story/14799

File: content/docs/docs-nav.yaml [hunk 1]
```diff
@@ -66,34 +66,40 @@ nav:
...
-      - Writing scaffolder templates: '/docs/details/scaffolder/'
-  - Scaffolder-Cookbook:
-      - Create RFC: '/docs/scaffolder-cookbook/create-rfc/'
-      - Debug template: '/docs/scaffolder-cookbook/debug-template/'
-      - Append file: '/docs/scaffolder-cookbook/append/'
-      - Re-use Parameters: '/docs/scaffolder-cookbook/global-constants/'
-      - Edit Files: '/docs/scaffolder-cookbook/edit-file/'
+  - Scaffolder:
+      - Writing scaffolder templates: '/docs/scaffolder/writing-templates/'
+      - Recipies:
```
Comment (Xantier) [inline content/docs/docs-nav.yaml:94]:
```suggestion
      - Recipes:
```

File: content/docs/docs-nav.yaml [hunk 2]
```diff
@@ -66,34 +66,40 @@ nav:
...
+        - Create RFC: '/docs/scaffolder/create-rfc/'
+        - Debug template: '/docs/scaffolder/debug-template/'
+        - Append file: '/docs/scaffolder/append/'
+        - Re-use Parameters: '/docs/scaffolder/global-constants/'
```
Comment (Xantier) [inline content/docs/docs-nav.yaml:98]:
```suggestion
        - Re-use parameters: '/docs/scaffolder/global-constants/'
```

Everything else apart from brands/names are lowercase

File: content/docs/docs-nav.yaml [hunk 3]
```diff
@@ -66,34 +66,40 @@ nav:
...
+        - Edit Files: '/docs/scaffolder/edit-file/'
```
Comment (Xantier) [inline content/docs/docs-nav.yaml:99]:
```suggestion
        - Edit files: '/docs/scaffolder/edit-file/'
```

**Review by Xantier — APPROVED** [review]

**Pull request merged**